### ISSUE TITLE
superfluous-parens wrongly detected with walrus operator in list comprehensions

### ISSUE THREAD

**Issue by jaysonsantos** [PyCQA/pylint#3383]
### Steps to reproduce
1. Create a file a.py
2. Paste this to it:
```python
def is_odd(n):
    return n % 2 != 0 and n

print([odd for i in range(100) if (odd := is_odd(i))])
```
3. run pylint a.py

### Current behavior
```
************* Module a
a.py:5:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)

------------------------------------------------------------------
```

### Expected behavior
This should not complain about the parenthesis because without it the syntax is invalid when using the walrus `:=` operator

### pylint --version output

```
pylint 2.4.4
astroid 2.3.3
Python 3.8.0 (default, Oct 17 2019, 17:08:29) 
[Clang 11.0.0 (clang-1100.0.33.8)]
```

**Comment by PCManticore** [comment]
Thanks for the report!

**Comment by OJFord** [comment]
I most often encounter this with negated conditions:
```python3
if not (two_is_odd := is_odd(2)):
    # Handle two not being odd
```

(In this case of course it could be fixed by moving `not` inside the expression, but only because it returns `bool`.)
